app/services/session_encryptor.rb

```diff
@@ -8,15 +8,13 @@ def self.build_user_access_key
     build_user_access_key
   end
 
-  # rubocop:disable Security/MarshalLoad
   def self.load(value)
     decrypted = encryptor.decrypt(value, user_access_key)
-    Marshal.load(::Base64.decode64(decrypted))
+    ActiveSupport::JSON.decode(::Base64.decode64(decrypted))
```

Comment:
Agreed about the migration issues. Rails 4.1 solved this by introducing a `hybrid` serializer that allowed Marshalled sessions to continue working while using JSON for new sessions. Once we are confident all users are on JSON, we can remove the hybrid code. I'll search the Rails codebase to see how they did the hybrid serializer.